client.go

```diff
@@ -20,6 +20,10 @@ type HAProxyClient struct {
 	conn net.Conn
 }
 
+type HAProxy interface {
+	RunCommand(cmd string) (*bytes.Buffer, error)
```

Comment:
Would it be easier to pass the `net.Conn` reference to the HAProxy struct and move the `dial` and `schema` functions into helper functions instead? 

This would decouple the implementation and external dependencies at the API boundary and could be done externally without having to export extra types.